### ISSUE TITLE
github sites

### ISSUE THREAD

**Issue by mokun** [goxr3plus/JavaFX-Web-Browser#5]
Hi @goxr3plus, 

Have you investigated why github sites won't work ?

e.g. if I type in `https://github.com/goxr3plus/JavaFX-Web-Browser`, it won't render correctly.

I'm using Java 9/10 64-bit in Windows

It almost looks like a crypto issue.

**Comment by goxr3plus** [comment]
My dear friend i currently stopped supporting this library because of
issues like that. The engine behind JavaFX WebBrowser is badly outdated. It
can't even play full hd youtube videos.

I think it has to do eith some background parsing....

:)

**Comment by goxr3plus** [comment]
I am working on another Browser right now based on Chromium engine.

Google seaech JxBrowser Java. I am building a JavaFX Browser on top of it.
OpenSource :)

**Comment by mokun** [comment]
is JxBrowser fully open source ? how frequent do they update the chromium engine ?

**Comment by mokun** [comment]
JavaFX's webengine is based on webkit, you know. Webkit is not that bad. 

It's just that JavaFX development has been very slow and they use older version of webkit and is forever playing catchup :(

I'm glad that beginning Java 11, the JavaFX portion will be handed to developers.

**Comment by goxr3plus** [comment]
Yes my friend it's amazing news with Java 11, i may start heavily updating
this web browser again. Until then JxBrowser is the solution at least for
me. It's almost exactly like Google Chrome, except the mp4 decodes etc
which you need to get a license too....

Check it here
https://www.teamdev.com/jxbrowser

Have also a look on XR3Player using it :
https://github.com/goxr3plus/XR3Player

**Comment by mokun** [comment]
But why are they charging license fee ?

https://www.teamdev.com/jxbrowser#licensing-pricing

> Evaluation Support 
> Provided for 30 days

Also, do they use webkit for video ?

> JxBrowser is using Webkit and FFmpeg components, supplied under LGPL.

**Comment by goxr3plus** [comment]
they just if you want to use it commercially, you know they have to survive
as a company.

Really i didn't knew they used WebKit.....

**Comment by goxr3plus** [comment]
Have a look at this StackOverflow question :


https://stackoverflow.com/questions/5850338/rendering-webpages-with-webkit-in-java

I hope now that JavaFX after separation from standart Java SDK becomes more
powerful and frequntly updated.


:)

**Comment by mokun** [comment]
We can use

`logger.info("Web Engine supported : " + engine.getUserAgent());`

to see what version of webkit the java jdk that we are using.

**Comment by mokun** [comment]
For JDK 8u131, it prints the following :

> Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/602.1 (KHTML, like Gecko) JavaFX/8.0 Safari/602.1

For JDK 10.0.2, 

> Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/605.1 (KHTML, like Gecko) JavaFX/10 Safari/605.1

**Comment by mokun** [comment]
I found this page : 

https://developers.whatismybrowser.com/useragents/explore/software_name/safari/?order_by=software_version

It's almost same tag info given by `engine.getUserAgent()` in JavaFX

or in here : 
https://developers.whatismybrowser.com/useragents/parse/667339-safari-macos-webkit

605.1 is used by Safari 11.1

**Comment by goxr3plus** [comment]
Awww now i catched it, thank you!!! :)

**Comment by goxr3plus** [comment]
Hello my friend , what should i do with this issue ? Soon i will pass this project on Java 11.
@mokun

**Comment by goxr3plus** [comment]
@mokun  Okay now i have the knowledge to pass this project on Java 11 .
